spec/index.bs

```diff
@@ -2928,8 +2919,13 @@ The {{GPUMapMode}} flags determine how a {{GPUBuffer}} is mapped when calling
                     Issue: Consider aligning mapAsync offset to 8 to match this.
                 </div>
 
-            1. Let |m| be a new [=mapped range ArrayBuffer=] of size |rangeSize| pointing at the content
-                of |this|.{{GPUBuffer/[[mapping]]}} at offset |offset| - |this|.{{GPUBuffer/[[mapping_range]]}}[0].
+            1. Let |m| be a new {{ArrayBuffer}} of size |rangeSize|, pointing at the content
+                at offset (|offset| - |this|.{{GPUBuffer/[[mapping_range]]}}[0])
+                of |this|.{{GPUBuffer/[[mapping]]}}.
+
+            1. Set |m|.{{ArrayBuffer/[[ArrayBufferDetachKey]]}} to "WebGPUBufferMapping".
+
+                Note: This prevents the ArrayBuffer from being detached by anything other than WebGPU.
```

Comment:
```suggestion
                Note: This prevents the ArrayBuffer from being detached by anything other than WebGPU, and trying to Transfer this ArrayBuffer will error.
```

Reply:
Actually I'll move the dev-targeted wording up out of the algorithm because I think algorithm details are too buried to be as accessible to devs.